homeassistant/helpers/condition.py

```diff
@@ -95,29 +98,33 @@ def async_from_config(
             )
         )
 
+    # Check for partials to properly determine if coroutine function
+    check_factory = factory
+    while isinstance(check_factory, ft.partial):
+        check_factory = check_factory.func
+
+    if asyncio.iscoroutinefunction(check_factory):
+        return await cast(Callable[..., bool], factory(hass, config, config_validation))
     return cast(Callable[..., bool], factory(config, config_validation))
 
 
 from_config = _threaded_factory(async_from_config)
```

Comment:
This is now broken because `async_from_config` is a coroutine function. Is this used anywhere?

Reply:
I think that we can remove all of those.

Reply:
OK, removed.